### ISSUE TITLE
Nested lazy execution problems

### ISSUE THREAD

**Issue by matthewowen** [rmosolgo/graphql-ruby#857]
I'm not certain if this is an issue with graphql-ruby of graphql-batch... but I think it's here.

Concretely, with a toy example:

```ruby
class BelongsToLoader < GraphQL::Batch::Loader
  def initialize(model)
    @model = model
  end

  def perform(ids)
    @model.where(id: ids).each { |record| fulfill(record.id, record) }
    ids.each { |id| fulfill(id, nil) unless fulfilled?(id) }
  end
end

ShipmentType = GraphQL::ObjectType.define do
  field :client, !ClientType do
    resolve -> (obj, _args, ctx) {
      BelongsToLoader.for(Client).load(obj.client_id)
    }
  end
end

ClientType = GraphQL::ObjectType.define do
  field :company, !ClientType do
    resolve -> (obj, _args, ctx) {
      BelongsToLoader.for(Company).load(obj.company_id)
    }
  end
end

CompanyType = GraphQL::ObjectType.define do
  field :primary_contact_user, !UserType do
    resolve -> (obj, _args, ctx) {
      BelongsToLoader.for(User).load(obj.primary_contact_user_id)
    }
  end
end

UserType = GraphQL::ObjectType.define do
  field :id, types.Int
end
```

With a schema like this and a viewer type, I can provider a query like 
```graphql
shipments {
  client {
    company {
      primary_contact_user {
        id
      }
    }
  } 
}
```

Now, the first two levels of association load as I'd hope: they're batched. But the bottom layer (loading the users) doesn't get batched properly. The `BelongsToLoader` is called, but `perform` is repeatedly called with just a single ID, so we do a query to get a user for every company.

This seems odd. As far as I can tell, graphql-batch. just hooks into the lazy resolution provided by graphql-ruby, so I think that's where the issue lies.

Obviously, this poses problems for more heavily nested schemas. I'd love some pointers on whether there's something I'm doing wrong, or whether this is expected, or whether it's a bug.

**Comment by rmosolgo** [comment]
Hi, thanks for reporting this issue! Are you on the latest graphql-ruby? I ask because a fix was just added in this area in #835

**Comment by matthewowen** [comment]
Nice! I upgraded and the issue is resolved. Thanks so much for the speedy response!
